openapi.yaml

```
@@ -471,6 +591,173 @@ paths:
           $ref: "#/components/responses/InternalServerError"
 components:
   schemas:
+    AddRelationArgs:
+      type: object
+      properties:
+        endpoints:
+          type: array
+          items:
+            type: string
+        via-cidrs:
+          type: array
+          items:
+            type: string
+      additionalProperties: false
+      required:
+        - endpoints
+    Application:
+      type: object
+      description: Information about a given application.
+      properties:
+        base:
+          $ref: '#/components/schemas/Base'
+        channel:
+          type: string
+        charm:
+          type: string
+        constraints:
+          $ref: '#/components/schemas/Constraints'
+        endpoint-bindings:
+          type: object
+          patternProperties:
+            .*:
+              type: string
```

Can we remove this `patternProperties` by either: modelling endpoint bindings as a list of well-defined objects, or more simply using the suggestion here.

```suggestion
          additionalProperties: true
```